File: src/test/java/org/hydev/mcpm/client/commands/presenter/TableTest.java
package org.hydev.mcpm.client.commands.presenter;

import org.hydev.mcpm.client.display.presenters.Table;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.hydev.mcpm.utils.ColorLogger.printc;

/**
 * Tests for FormatUtils
 */
class TableTest
{
    @Test
    void tabulate()
    {
        printc(Table.tabulate(List.of(
            List.of("Azalea", "meow", "Cakes without eggs or sugar"),
            List.of("Lindsey", "meow", "Burgers")
        ), List.of(":Contributor", "Centered", "Favorite Food:")));
    }
}

File: src/test/java/org/hydev/mcpm/client/commands/presenter/TableTest.kt
package org.hydev.mcpm.client.commands.presenter

import org.hydev.mcpm.client.display.presenters.Table
import org.hydev.mcpm.utils.ColorLogger.printc
import org.junit.jupiter.api.Test

/**
 * Tests for FormatUtils
 */
internal class TableTest
{
    @Test
    fun tabulate()
    {
        printc(Table(listOf(":Contributor", "Centered", "Favorite Food:"), listOf(
            listOf("Azalea", "meow", "Cakes without eggs or sugar"),
            listOf("Lindsey", "meow", "Burgers")
        )).toString())
    }
}
